daemon/cluster/services.go

```diff
@@ -134,6 +138,35 @@ func (c *Cluster) CreateService(s types.ServiceSpec, encodedAuth string, queryRe
 
 		switch serviceSpec.Task.Runtime.(type) {
 		// handle other runtimes here
+		case *swarmapi.TaskSpec_Generic:
+			switch serviceSpec.Task.GetGeneric().Kind {
+			case string(types.RuntimePlugin):
+				if s.TaskTemplate.PluginSpec == nil {
+					return errors.New("plugin spec must be set")
+				}
+				p := runtime.GenericPluginPayload{
+					Spec: *s.TaskTemplate.PluginSpec,
+				}
+				if encodedAuth != "" {
+					p.PullOptions = runtime.PluginPullOptions{RegistryAuth: encodedAuth}
```

Comment:
There's a hack in the `ContainerSpec` version of the code that carries the registry auth credentials out-of-band to avoid exposing them to API clients. I'm not sure this was a good design choice; it's caused us some pain. It may not make sense to carry it forward for the `Plugin` runtime. Re-marshalling the protobuf here isn't great - among other things, it means any new fields that this implementation doesn't understand will be stripped out. However, what this is doing now is most consistent with `ContainerSpec`.

@aluzzardi: WDYT?